content/news/2022-11-05-bevy-0.9/index.md

```diff
@@ -0,0 +1,1045 @@
+  }
+}
+
+// New
+fn some_system(world: &mut World, transforms: &mut QueryState<&Transform>) {
+  for transform in transforms.iter(world) {
+  }
+}
+```
+
+[`Local`] can be used to store local data inside of the exclusive system:
+
+```rust
+// Old
+#[derive(Resource)]
+struct Counter(usize);
+fn some_system(world: &mut World) {
+  let mut counter = world.resource_mut::<Counter>();
+  counter.0 += 1;
+}
+
+// New
+fn some_system(world: &mut World, mut counter: Local<usize>) {
+  *counter += 1;
+}
+```
+
+[`World`]: https://docs.rs/bevy/0.9.0/bevy/ecs/world/struct.World.html
+[`QueryState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/struct.QueryState.html
+[`SystemState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.SystemState.html
+[`Local`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.Local.html
+
+## Bevy ECS Now Uses GATS!
+
+<div class="release-feature-authors">authors: @BoxyUwU</div>
+
+Rust 1.65.0 [stabilized GATs](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html) (Generic Associated Types), which enabled us to significantly simplify Bevy ECS Query internals.
+
+For awhile now, Bevy ECS has been hacking around the lack of GATs with a complicated nest of traits (`WorldQuery`, `WorldQueryGats` (a "lack of real GATs hack" trait), and `Fetch`).
+
+In **Bevy 0.9**, we now have a single [`WorldQuery`] trait! This makes Bevy ECS much easier to maintain, extend, debug, document, and understand.
+
+[`WorldQuery`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/trait.WorldQuery.html
+
+## Derive Resource
+
+<div class="release-feature-authors">authors: @devil-ira</div>
+
+The [`Resource`] trait is now no longer automatically implemented for all types. It must be derived:
+
+```rust
+#[derive(Resource)]
+struct Counter(usize);
+```
+
+This change was made on the tail of [making the same decision](/news/bevy-0-6/#the-new-component-trait-and-derive-component) for [`Component`] types. In short:
+
+1. Auto-implementing [`Resource`] for every type made it very easy to accidentally insert the "wrong" value, such as inserting the constructor function pointer instead of the value itself:
+
+    ```rust
+    struct Counter(usize);
+    // This inserts the constructor function pointer as a resource!
+    // Weird and confusing!
+    app.insert_resource(Counter);
+    ```
+
+2. Deriving [`Resource`] documents intent in a structured way. Without a derive, resource-ness is implicit by default.
+3. Auto-implementing meant that plugins could use the same "common" type in conflicting ways (ex: `std` types like `Vec<usize>`). Not implementing by default means that plugins cannot use these common types in conflicting ways. They must create new types.
+4. This opens the door to configuring resource types using the Rust type system (like we already do for components).
+
+[`Resource`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/trait.Resource.html
+
+## Query Internals Reworks
+
+- [make `WorldQuery` very flat][5205]
```

Comment:
This PR was already in 0.8 I think, if you look at docs for 0.8 you'll see that `WorldQueryGats` is a thing

Reply:
These changes were merged on August 4th, but Bevy 0.8 released on July 30th.
WorldQueryGats existed prior to 5205.

Reply:
Oh... wild, I thought that worldquerygats was introduced at the same time as moving everything onto `WorldQuery` :scream: nvm me then